### PR TITLE
TR-1284: Create next template editor page

### PR DESCRIPTION
**Pull request by bkemper**
Refer to [TR-1284](https://jira.int.messagesystems.com/browse/TR-1284) for AC.  Please note this is a subtask.

### What Changed

- New page for next template editor

Screen Shot 2019-05-10 at 5 34 15 PM

### How To Test
 - Visit http://app.sparkpost.test/templates/edit/amp/next

### CODE REVIEW

**Comment by bkemper** [comment]
@rajumsys This PR is ready for your 👁 👁 when you have time.

File: src/context/RouterContext.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React, { createContext, useMemo } from 'react';
+import { withRouter } from 'react-router';
+import qs from 'qs';
+
+// see, https://github.com/ReactTraining/react-router/issues/6430
+// note, maintainers do not recommend using __RouterContext from react-router
+const RouterContext = createContext();
+
+export const ProviderComponent = ({ children, ...routerProps }) => {
+  const value = useMemo(() => ({
```
Comment (rajumsys) [inline src/context/RouterContext.js]: is `useMemo` useful w/o args?

File: src/pages/templates/EditAndPreviewPage.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import { Grid, Page } from '@sparkpost/matchbox';
+import useRouter from 'src/hooks/useRouter';
+
+const EditAndPreviewPage = () => {
+  const router = useRouter();
+
+  return (
+    <Page
+      breadcrumbAction={{ content: 'Back Link', to: '/templates' }}
+      title={router.requestParams.id}
```
Comment (rajumsys) [inline src/pages/templates/EditAndPreviewPage.js]: `.name` (show template name)

File: src/pages/templates/EditAndPreviewPage.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { Grid, Page } from '@sparkpost/matchbox';
+import useRouter from 'src/hooks/useRouter';
+
+const EditAndPreviewPage = () => {
+  const router = useRouter();
```
Comment (rajumsys) [inline src/pages/templates/EditAndPreviewPage.js]: Is this a pattern we intend to implement going forward?

File: src/pages/templates/EditAndPreviewPage.js (hunk 2 above)
Comment (bkemper) [reply src/pages/templates/EditAndPreviewPage.js]: In TR-1284, I fetch the template and set the title to the template name.

File: src/pages/templates/EditAndPreviewPage.js (hunk 3 above)
Comment (bkemper) [reply src/pages/templates/EditAndPreviewPage.js]: I think so.  The benefit verse the HOC is readability and a single place to abstract common router needs, such as, requestParams.  Let's see how it performs on this prototype.

File: src/context/RouterContext.js (hunk 1 above)
Comment (bkemper) [reply src/context/RouterContext.js]: I think Kent said it best.

> Most of the time, this wont be a huge deal, especially if your context value doesn't change very often, but it's normally pretty easy to memoize the value with React.useMemo anyway so may as well do it all the time for your context value and avoid the problem.
> — https://kentcdodds.com/blog/how-to-optimize-your-context-value

File: src/pages/templates/EditAndPreviewPage.js (hunk 2 above)
Comment (rajumsys) [reply src/pages/templates/EditAndPreviewPage.js]: ah, that's right. name isn't available yet.

File: src/context/RouterContext.js (hunk 1 above)
Comment (rajumsys) [reply src/context/RouterContext.js]: haha. it's ok

_thinking about optimization w/o measuring :)_

File: src/hooks/useRouter/useRouter.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { useContext } from 'react';
+import RouterContext from 'src/context/RouterContext';
+
+// see, https://github.com/ReactTraining/react-router/issues/6430
+// watch, https://www.youtube.com/watch?v=3yiialslPbc
+// note, custom hooks like this one are recommended to make mocking in tests easier
+const useRouter = () => useContext(RouterContext);
```
Comment (rajumsys) [inline src/hooks/useRouter/useRouter.js]: a quick question: could we just use the context in the component? what's the benefit of using this custom hook when it doesn't use any other hooks/effects?

**Pull request merged**